libs/community/langchain_community/document_loaders/pebblo.py

```diff
@@ -158,11 +160,16 @@ def _classify_doc(self, loading_end: bool = False) -> dict:
         for doc in doc_content:
             doc_metadata = doc.get("metadata", {})
             doc_authorized_identities = doc_metadata.get("authorized_identities", [])
-            doc_source_path = get_full_path(
-                doc_metadata.get(
-                    "full_path", doc_metadata.get("source", self.source_path)
+            if isinstance(self.loader, SharePointLoader):
```

Comment:
For safety, compare class names. If the community decides to move SharePointLoader into a new repo as was done for GoogleDriveLoader, this will not work. 

Ref: https://github.com/dristysrivastava/langchain/blob/ee96e16f9ac3fa48ffb7b9e22033851b5b6e04a3/libs/community/langchain_community/utilities/pebblo.py#L233

Reply:
Updated